source/controllers/include/controllers/Controller.hpp

```diff
@@ -0,0 +1,72 @@
+/**
+ * @author Baptiste Busch
+ * @date 2021/02
+ *
+ */
+
+#pragma once
+
+#include "controllers/exceptions/NotImplementedException.hpp"
+#include "state_representation/Parameters/ParameterInterface.hpp"
+#include <list>
+#include <memory>
+
+namespace controllers {
+/**
+ * @class Controller
+ * @brief Abstract class to define a controller either in joint or cartesian spaces
+ * @tparam SIn the input state type of the controller
+ * @tparam SOut the output command type of the controller
+ */
+template <class SIn, class SOut>
+class Controller {
+public:
+  /**
+   * @brief Empty constructor
+   */
+  explicit Controller();
+
+  /**
+   * @brief Compute the command based on the input state
+   * To be redefined based on the actual controller implementation
+   * @param state the input state of the system. This function accept any number of extra arguments.
+   * @return the output command at the input state
+   */
+  virtual SOut compute_command(const SIn& state, ...) const;
+
+  /**
+   * @brief Compute the command based on the desired state and a feedback state
+   * To be redefined based on the actual controller implementation.
+   * This function accept any number of extra arguments.
```

Comment:
No ellipsis in function declaration, so either this comment line is wrong or you meant to allow variadic argument below

Reply:
Yes I was not sure with that. I think we actually need an example where it makes sense to have those variadic arguments and at the end I could not find so I removed it in this declaration but forgot the comment.